File: HelperJobby/DTOs/Account/JobSeekerDTO.cs
using HelperJobby.DTOs.Address;
using HelperJobby.DTOs.Resume;
using HelperJobby.DTOs.User;
using HelperJobby.DTOs.UserJobInteractions;

namespace HelperJobby.DTOs.Account;

public class JobSeekerDTO
{
    public int Id { get; set; }

    public string FirstName { get; set; }

    public string LastName { get; set; }
    public string PhoneNumber { get; set; }

    public int UserId { get; set; }

    public UserDTO User { get; set; }

    public int? AddressId { get; set; }
    public AddressDTO? Address { get; set; }
    public ResumeDTO? Resume { get; set; }
    public List<InterviewDTO> Interviews { get; set; }
    public List<JobApplyDTO> JobApplies { get; set; }
    public List<SavedJobDTO> SavedJobs { get; set; }
}

File: HelperJobby/client-app/src/DTOs/accountDTOs/JobSeekerDTO.ts
import {AddressDTO} from "../addressDTOs/AddressDTO";
import {ResumeDTO} from "../resumeRelatedDTOs/ResumeDTO";
import {JobApplyDTO} from "../userJobInteractionsDTOs/JobApplyDTO";
import {SavedJobDTO} from "../userJobInteractionsDTOs/SavedJobDTO";
import {InterviewDTO} from "../userJobInteractionsDTOs/InterviewDTO";
import {UserDTO} from "../userRelatedDTOs/UserDTO";
import {ConversationDTO} from "../MessagingDTOs/ConversationDTO";

export interface JobSeekerDTO {
    id: number;
    firstName: string;
    lastName: string;
    phoneNumber: string;
    userId: number;
    user: UserDTO;
    addressId: number | null;
    address: AddressDTO | null;
    resume: ResumeDTO | null;
    interviews: InterviewDTO[];
    jobApplies: JobApplyDTO[];
    savedJobs: SavedJobDTO[];
    conversations: ConversationDTO[];
}